### ISSUE TITLE
🔥 Could not find firebase-iid-interop.aar (google() already on top of jcenter())

### ISSUE THREAD

**Issue by rodrigoreis22** [invertase/react-native-firebase#1660]
## Issue

I'm updating my app now after some time without touching it and I'm getting an error while compiling the android version:
```
Error:FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Could not find firebase-iid-interop.aar (com.google.firebase:firebase-iid-interop:16.0.0).
  Searched in the following locations:
      https://jcenter.bintray.com/com/google/firebase/firebase-iid-interop/16.0.0/firebase-iid-interop-16.0.0.aar
> Could not find play-services-basement.aar (com.google.android.gms:play-services-basement:15.0.1).
  Searched in the following locations:
      https://jcenter.bintray.com/com/google/android/gms/play-services-basement/15.0.1/play-services-basement-15.0.1.aar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```

---

#### `android/build.gradle`:

```groovy
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.3'
        classpath 'com.google.gms:google-services:4.0.1'
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven {
            url 'https://maven.google.com/'
            name 'Google'
        }
    }
}
```

#### `android/app/build.gradle`:

```groovy
apply plugin: "com.android.application"

import com.android.build.OutputFile

project.ext.react = [
    entryFile: "index.js"
]

project.ext.envConfigFiles = [
        debug: ".env",
        devrelease: ".env",
        release: ".env.prd"
]

apply from: "../../node_modules/react-native/react.gradle"
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 26
    buildToolsVersion '26.0.1'

    dexOptions {
        jumboMode = true
        javaMaxHeapSize "4g"
    }

    defaultConfig {
        resValue "string", "build_config_package", "com.arenaapp.publisher"
        applicationId "com.arenaapp.publisher"
        minSdkVersion 16
        targetSdkVersion 22
        multiDexEnabled  true
        versionCode 342
        versionName "1.4.1"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        debug {
            applicationIdSuffix ".dev"
            resValue "string", "app_name", "Arena.im.DEV"
        }
        devrelease {
            applicationIdSuffix ".dev"
            resValue "string", "app_name", "Arena.im.DEV"
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
        release {
            resValue "string", "app_name", "Arena.im"
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation project(':react-native-firebase')
    implementation project(':react-native-intercom')
    implementation project(':react-native-mixpanel')
    implementation project(':react-native-audio-toolkit')
    implementation project(':react-native-twitter')
    implementation project(':react-native-prompt-android')
    implementation project(':react-native-fbsdk')
    implementation project(':bugsnag-react-native')
    implementation project(':react-native-thumbnail')
    implementation project(':react-native-video')
    implementation project(':react-native-fetch-blob')
    implementation project(':react-native-camera')
    implementation project(':react-native-svg')
    implementation project(':react-native-config')
    implementation project(':react-native-i18n')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:26.0.1"
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation 'io.intercom.android:intercom-sdk-base:3.+'
    implementation "com.google.android.gms:play-services-base:11.6.0"
    // Firebase dependencies
    implementation "com.google.android.gms:play-services-base:15.0.1"
    implementation "com.google.firebase:firebase-core:16.0.3"
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

allprojects {
    repositories {
        // React Native default repository
        maven { url "$rootDir/../node_modules/react-native/android" }

        maven { url 'https://maven.google.com' }
    }
}

apply plugin: 'com.google.gms.google-services'
```

#### `android/settings.gradle`:

```groovy
rootProject.name = 'ArenaPublisher'
include ':react-native-firebase'
project(':react-native-firebase').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-firebase/android')
include ':react-native-intercom'
project(':react-native-intercom').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-intercom/android')
include ':react-native-mixpanel'
project(':react-native-mixpanel').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-mixpanel/android')
include ':react-native-twitter'
project(':react-native-twitter').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-twitter/android')
include ':react-native-prompt-android'
project(':react-native-prompt-android').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-prompt-android/android')
include ':react-native-fbsdk'
project(':react-native-fbsdk').projectDir = new File(rootProject.projectDir, '../libs/react-native-fbsdk/android')
include ':bugsnag-react-native'
project(':bugsnag-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/bugsnag-react-native/android')
include ':react-native-thumbnail'
project(':react-native-thumbnail').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-thumbnail/android')
include ':react-native-video'
project(':react-native-video').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-video/android')
include ':react-native-fetch-blob'
project(':react-native-fetch-blob').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-fetch-blob/android')
include ':react-native-camera'
project(':react-native-camera').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-camera/android')
include ':react-native-svg'
project(':react-native-svg').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-svg/android')
include ':react-native-config'
project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
include ':react-native-i18n'
project(':react-native-i18n').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-i18n/android')
include ':react-native-audio-toolkit'
project(':react-native-audio-toolkit').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-audio-toolkit/android/lib')

include ':app'
```

#### `MainApplication.java`:

```java
package com.arenaapp.publisher;

import android.app.Application;

import com.facebook.react.ReactApplication;
import io.invertase.firebase.RNFirebasePackage;
import com.robinpowered.react.Intercom.IntercomPackage;
import io.intercom.android.sdk.Intercom;
import com.kevinejohn.RNMixpanel.RNMixpanel;
import ga.piroro.rnt.RNTPackage;
import im.shimo.react.prompt.RNPromptPackage;
import io.invertase.firebase.auth.RNFirebaseAuthPackage;
import io.invertase.firebase.database.RNFirebaseDatabasePackage;

import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.bugsnag.BugsnagReactNative;
import com.reactlibrary.RNThumbnailPackage;
import com.brentvatne.react.ReactVideoPackage;
import com.RNFetchBlob.RNFetchBlobPackage;
import com.lwansbrough.RCTCamera.RCTCameraPackage;
import com.horcrux.svg.SvgPackage;
import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;
import com.AlexanderZaytsev.RNI18n.RNI18nPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.futurice.rctaudiotoolkit.AudioPackage;

import java.util.Arrays;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {

	private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new RNFirebasePackage(),
            new RNFirebaseDatabasePackage(),
            new RNFirebaseAuthPackage(),
            new IntercomPackage(),
            new RNMixpanel(),
            new RNTPackage(),
            new RNPromptPackage(),
            BugsnagReactNative.getPackage(),
						new FBSDKPackage(mCallbackManager),
            new RNThumbnailPackage(),
            new ReactVideoPackage(),
            new RNFetchBlobPackage(),
            new RCTCameraPackage(),
            new SvgPackage(),
            new ReactNativeConfigPackage(),
            new RNI18nPackage(),
						new AudioPackage()
      );
    }

    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

	protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    BugsnagReactNative.start(this);
    SoLoader.init(this, /* native exopackage */ false);
    FacebookSdk.sdkInitialize(getApplicationContext());
  	AppEventsLogger.activateApp(this);
    Intercom.initialize(this, "android_sdk-redacted", "redacted");
  }
}
```

#### `AndroidManifest.xml`:

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.arenaapp.publisher"
	xmlns:tools="http://schemas.android.com/tools"
    android:versionCode="342"
    android:versionName="1.4.1">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
	<uses-permission android:name="android.permission.CAMERA" />
	<uses-permission android:name="android.permission.RECORD_AUDIO"/>
  	<uses-permission android:name="android.permission.RECORD_VIDEO"/>
  	<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
  	<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />

    <uses-sdk
        android:minSdkVersion="16"
        android:targetSdkVersion="22" />

    <application
      android:name=".MainApplication"
      android:allowBackup="true"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:theme="@style/AppTheme">
				<meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/facebook_app_id"/>
        <activity
            android:name=".SplashActivity"
            android:label="@string/app_name"
            android:exported="true"
            android:theme="@style/SplashTheme">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <activity
            android:name=".MainActivity"
            android:label="@string/app_name"
            android:exported="true"
            android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
						android:launchMode="singleTask"
						android:screenOrientation="portrait"
            android:windowSoftInputMode="adjustResize">

					<intent-filter>
            <action android:name="android.intent.action.VIEW" />
            <category android:name="android.intent.category.DEFAULT" />
            <category android:name="android.intent.category.BROWSABLE" />
            <data android:scheme="arenapublisher" />
        	</intent-filter>
        </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
      <meta-data android:name="com.bugsnag.android.API_KEY" android:value="e61bc687e9ca14c2c0b7d23fb5d5b6b8"/>
			<service
     		android:name="com.robinpowered.react.Intercom.IntercomIntentService"
     		android:exported="false">
     		<intent-filter
        	android:priority="999">
        	<action android:name="com.google.android.c2dm.intent.RECEIVE"/>
    		</intent-filter>
			</service>
 			<receiver
     		android:name="io.intercom.android.sdk.push.IntercomPushBroadcastReceiver"
     		tools:replace="android:exported"
     		android:exported="true" />
    </application>

</manifest>
```

---

## Environment

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **If known, the version of the platform are you experiencing the issue on:**
  - Android API 26
- **Operating System:**
  - [x] MacOS, version: `10.13.6 High Sierra`
  - [ ] Windows, version: `N/A`
  - [ ] Other, please specify: `N/A`
- **Build Tools:**
  - `Android Studio 3.0.1`
- **`React Native` version:**
  - `react-native-cli: 2.0.1 / react-native: 0.57.1`
- **`React Native Firebase` library version:**
  - `5.0.0`
- **`Firebase` module(s) you're using that has the issue:**
  - [x] **N/A**
  - [x] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [ ] Cloud **Messaging** (FCM)
  - [ ] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] Invites
  - [ ] Instance ID
  - [ ] Notifications
  - [ ] Performance Monitoring
  - [x] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [x] No
  - [ ] Yes, version: `N/A`
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

**Comment by mkchx** [comment]
Hi, @rodrigoreis22 
Try to add the `google()` inside the **allprojects.repositories**, above **mavenLocal**
and see if it builds

**Comment by rodrigoreis22** [comment]
After doing what @mkchx suggested and also many other updates I was able to have a successful build. But after I run on simulator, this error is displayed:

```
Java exception in 'NativeModules'

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/database/ServerValue;
io.invertase.firebase.database.RNFirebaseDatabase.getConstants
    RNFirebaseDatabase.java:841
com.facebook.react.bridge.JavaModuleWrapper.getConstants
    JavaModuleWrapper.java:138
android.os.Handler.handleCallback
    Handler.java:751
android.os.Handler.dispatchMessage
    Handler.java:95
com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage
    MessageQueueThreadHandler.java:29
SyncTree
    SyncTree.js:16:22
<unknown>
    SyncTree.js:313:15
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    Reference.js:12
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    index.js:6
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    app.js:11
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    apps.js:2
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    firebase.js:2
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    index.js:1
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    firebase-api.js:1
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    delete-post-by-id.js:4
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    index.js:3
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    store.js:9
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:186:36
metroRequire
    require.js:115:90
<unknown>
    index.js:13
loadModuleImplementation
    require.js:292:12
guardedLoadModule
    require.js:179:45
metroRequire
    require.js:115:90
global code
    
```

https://d.pr/i/S6BXpS

**Comment by mkchx** [comment]
There's a conflict with your dependencies,
you're referencing **com.google.android.gms:play-services-base** two times
in android/app/build.gradle:
```
implementation "com.google.android.gms:play-services-base:11.6.0"
// Firebase dependencies
implementation "com.google.android.gms:play-services-base:15.0.1"
```
Remove the duplicate and change the version to **16.0.1** e.g
`implementation "com.google.android.gms:play-services-base:16.0.1"`

Also change the version of **com.google.firebase:firebase-core** to **16.0.4** e.g
`implementation "com.google.firebase:firebase-core:16.0.4"`

And you miss a dependency, since you want to use the **firebase-database**, add
 `implementation "com.google.firebase:firebase-database:16.0.3"`

**Comment by rodrigoreis22** [comment]
@mkchx thanks.

I've tried that now but now I'm seeing this error:
```
The library com.google.android.gms:play-services-base is being requested by various other libraries at [[15.0.1,15.0.1]], but resolves to 16.0.1. Disable the plugin and check your dependencies tree using ./gradlew :app:dependencies.
```

Now it's weird that I changed back to:
`implementation "com.google.android.gms:play-services-base:15.0.1"` 

and still the same message. I've tried cleaning the project, closing android studio...

**Comment by cebondlovu** [comment]
Just update
